os_migrate/plugins/module_utils/resource.py

```diff
@@ -0,0 +1,236 @@
+        set_ser_params_same_name(info, sdk_res, self.info_from_sdk)
+        set_ser_params_same_name(info, refs, self.info_from_refs)
+
+    def _sdk_params_from_params_and_refs(self, sdk_params, refs):
+        # we could move these methods from serialization.py to private
+        # under Resource
+        set_sdk_params_same_name(self.params(), sdk_params, self.sdk_params_from_params)
+        set_sdk_params_same_name(refs, sdk_params, self.sdk_params_from_refs)
+
+    def type(self):
+        return self.data[const.RES_TYPE]
+
+    def params(self):
+        return self.data[const.RES_PARAMS]
+
+    def info(self):
+        return self.data[const.RES_INFO]
+
+    def params_and_info(self):
+        return (self.params(), self.info())
+
+    def _sort_param(self, param_name):
+        self.params()[param_name] = sorted(self.params()[param_name])
+
+    def _sort_param(self, param_name):
+        self.info()[param_name] = sorted(self.info()[param_name])
+
+
+# This would obviously go to network.py but i want to keep the "idea
+# pull request" simple.
+class Network(Resource):
+
+    params_from_sdk = [
+        'description',
+        'dns_domain',
+        'is_admin_state_up',
+        'is_default',
+        'is_port_security_enabled',
+        'is_router_external',
+        'is_shared',
+        'is_vlan_transparent',
+        'mtu',
+        'name',
+        'provider_network_type',
+        'provider_physical_network',
+        'provider_segmentation_id',
+        'segments',
+    ]
+    params_from_refs = [
+        'qos_policy_name',
+    ]
+    info_from_sdk = [
+        'availability_zones',
+        'created_at',
+        'id',
+        'project_id',
+        'qos_policy_id',
+        'revision_number',
+        'status',
+        'updated_at',
+    ]
+
+    sdk_params_from_params = [
+        'description',
+        'dns_domain',
+        'is_admin_state_up',
+        'is_default',
+        'is_port_security_enabled',
+        'is_router_external',
+        'is_shared',
+        'is_vlan_transparent',
+        'mtu',
+        'name',
+        'provider_network_type',
+        'provider_physical_network',
+        'provider_segmentation_id',
+        'segments',
+    ]
+    sdk_params_from_refs = [
+        'qos_policy_id',
+    ]
+
+
+    @classmethod
+    def from_sdk(cls, sdk_resource, conn):
+        # >>> Internal composition over inheritance! <<<
+        # This is more future-proof, composable, less restrictive
+        # approach than calling the same method via super.
+        obj = cls()
+        obj._data_from_sdk_and_refs(sdk_resource, _refs_from_sdk(sdk_resource, conn))
+        # Let's not make special lists for sorted params/info, just sort them?
+        obj._sort_param('availability_zone_hints')
+        obj._sort_info('subnet_ids')
+        return obj
+
+    ## TBD: SOME MORE DRY HERE? Can we refactor something more into reusable bits?
+    def import_or_update(self, conn):
```

Comment:
not material to the review, but everything else in openstacksdk is "create" or "update".  I'd like "create_or_update" here for the method name.

Reply:
Done